cmd/eventhandling/event_processor.go

```diff
@@ -0,0 +1,135 @@
+package eventhandling
+
+import (
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/event"
+	events "github.com/keptn-sandbox/echo-service/pkg"
+	keptnv2 "github.com/keptn/go-utils/pkg/lib/v0_2_0"
+	"log"
+)
+
+// CloudEventProcessor defines a processor on cloud events
+type CloudEventProcessor interface {
+	Process(event cloudevents.Event) error
+}
+
+// EchoCloudEventProcessor is the default implementation of a CloudEventProcessor that hhah
+type EchoCloudEventProcessor struct {
+	EventSender EventSender
+	Sleeper     Sleeper
+}
+
+// BrokenEchoCloudEventProcessor is an implementation that does wrong event signalling. I.e., it
+// sends a .finished event before a .started event.
+// Useful for testing purposes.
+type BrokenEchoCloudEventProcessor struct {
+	EventSender EventSender
+	Sleeper     Sleeper
+}
+
+// Process processes a cloud event
+func (ep EchoCloudEventProcessor) Process(event cloudevents.Event) error {
+
+	if event.Type() == events.EchoEventTriggeredType {
+		log.Println("Processing Echo Triggered Event")
+		eventData := &events.EchoTriggeredEventData{}
+		if err := event.DataAs(eventData); err != nil {
+			log.Printf("Got Data Error: %s", err.Error())
+			return err
+		}
+
+		if err := ep.EventSender.SendEvent(createEchoStartedEvent(event)); err != nil {
+			log.Printf("Got Send Error: %s", err.Error())
+			return err
+		}
+
+		log.Printf("GOT EVENT :)\n")
```

Comment:
Should we also print the actual event that we got here? Or should we remove this log statement?

Reply:
let's print the event type, done